MapboxNavigation/CarPlaySearchManager+Search.swift

```diff
@@ -149,12 +132,27 @@ extension CarPlayManager: CPSearchTemplateDelegate {
     }
 }
 
+@available(iOS 12.0, *)
+extension CarPlaySearchManager: CPListTemplateDelegate {
+    public func listTemplate(_ listTemplate: CPListTemplate, didSelect item: CPListItem, completionHandler: @escaping () -> Void) {
+        // Selected a search item from the extended list?
+        if let userInfo = item.userInfo as? [String: Any],
+            let placemark = userInfo[CarPlaySearchManager.CarPlayGeocodedPlacemarkKey] as? GeocodedPlacemark,
+            let location = placemark.location {
+            let destinationWaypoint = Waypoint(location: location)
+            delegate?.popTemplate(animated: false)
+            delegate?.previewRoutes(between: [destinationWaypoint], completionHandler: completionHandler)
```

Comment:
these delegates appear to be nil, causing the flow to stall when selecting a `CPListItem`.

Reply:
The wrong function was called here. I corrected it in this new [commit](https://github.com/mapbox/mapbox-navigation-ios/pull/1846/commits/4a5d7d5118972d069e22efd4896b4f70476b3462).